yt_idefix/data_structures.py

```diff
@@ -320,3 +321,98 @@ def _parse_parameter_file(self):
 class PlutoVtkDataset(IdefixVtkDataset):
     _version_regexp = re.compile(r"\d+\.\d+\.?\d*[-\w+]*")
     _required_header_keyword = "PLUTO"
+
+    def __init__(
+        self,
+        filename,
+        dataset_type=None,
+        unit_system="cgs",
+        units_override=None,
+        *,
+        code_dir: str | None = None,
```

Comment:
Unless you have ideas how this argument could be used for other things in the future, I'd much prefer exposing a `definitions_header` argument directly, which would match how the `inifile` argument is implemented. The corresponding instance attribute should be "hidden" rather than public (`self._definitions_header` instead of `self.code_dir`)

Reply:
make sense! I'll change it